### PR TITLE
Skip OCP Route Updates for Local Environment Work

### PR DESCRIPTION
**Pull request by NithinKuruba**
# Description

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (non-breaking change with enhancements to documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: microservices/gatewayApi/v2/routes/gateway.py [hunk 1]
```diff
@@ -77,28 +78,30 @@ def delete_config(namespace: str, qualifier = "") -> object:
...
+                    if check_nsp (namespace, ocp_ns) is False:
+                        apply_nsp (namespace, ocp_ns, tempFolder)
 
             # ok all looks good, so update a secret containing the original submitted request
             log.debug("%s - Update Original Config" % (namespace))
             write_submitted_config ("", tempFolder)
-            prep_and_apply_secret (namespace, selectTag, tempFolder)
+            if not local_environment:
+                prep_and_apply_secret (namespace, selectTag, tempFolder)
             log.debug("%s - Updated Original Config" % (namespace))
```
Comment (ikethecoder) [inline microservices/gatewayApi/v2/routes/gateway.py:105]: The highlighted lines are actually all related.. so can be part of the `if not local_environment`.

File: microservices/gatewayApi/v2/routes/gateway.py [hunk 2]
```diff
@@ -273,29 +276,32 @@ def write_config(namespace: str) -> object:
...
+                        if check_nsp (namespace, ocp_ns) is False:
+                            apply_nsp (namespace, ocp_ns, tempFolder)
 
             # ok all looks good, so update a secret containing the original submitted request
             log.debug("[%s] - Update Original Config" % (namespace))
             write_submitted_config (orig_config, tempFolder)
-            prep_and_apply_secret (namespace, selectTag, tempFolder)
+            if not local_environment:
+                prep_and_apply_secret (namespace, selectTag, tempFolder)
             log.debug("[%s] - Updated Original Config" % (namespace))
```
Comment (ikethecoder) [inline microservices/gatewayApi/v2/routes/gateway.py:305]: The highlighted lines are actually all related.. so can be part of the `if not local_environment`.

**Review by ikethecoder — CHANGES_REQUESTED** [review]

File: microservices/gatewayApi/v2/routes/gateway.py (hunk 1 above)
Comment (NithinKuruba) [reply microservices/gatewayApi/v2/routes/gateway.py]: @ikethecoder Updated the code and grouped all those lines under single if condition.

File: microservices/gatewayApi/v2/routes/gateway.py (hunk 2 above)
Comment (NithinKuruba) [reply microservices/gatewayApi/v2/routes/gateway.py]: @ikethecoder Updated the code and grouped all those lines under single if condition.

**Review by ikethecoder — APPROVED** [review]

**Pull request merged**